petabvis/plot_row.py

```diff
@@ -68,10 +69,12 @@ def get_simulation_condition_id(self):
         if len(simulation_condition_id) == 1:
             return simulation_condition_id[0]
         else:
-            raise NotImplementedError("The Simulation Condition ID should"
-                                      "be unique for a line and between"
-                                      "replicates")
-
+            warnings.warn("The Simulation Condition ID should "
+                          "be unique for a line and between replicates")
+            return "Simulation Condition ID not unique for line"
+            # raise NotImplementedError("The Simulation Condition ID should "
+            #                         "be unique for a line and between "
+            #                         "replicates")
```

Comment:
```suggestion
            warnings.warn("The Simulation Condition ID should "
                          "be unique for a line and between replicates")
            return "Simulation Condition ID not unique for line"
```

Reply:
Should the return be changed to `return None`? I think this method is only called at line 36 of this file, and there I think assigning this string to that variable doesn't make much sense.

Reply:
This id can also be used to color the points in the correlation plot and is displayed when clicking on a point in the correlation plot. I think it may be nice to give a somewhat descriptive name in this case because otherwise the user might be confused when he can color the plot by a "None" ID and might think that something with his data could be wrong. This way, he can see exactely which points are affected by the warning message.
Also, I think that returning None here could lead to errors somewhere down the line.

Reply:
Sounds reasonable! Thanks for the explanation.

As an aside, I misunderstood "unique" here. Is the meaning equivalent to: "Error: replicates do not have the same simulation condition."? If so, it would be good to change the warning and return string to something like this.

Might be helpful to also add some information to the warning message, such that users can quickly find the issue in their PEtab files (e.g. which replicate ID has replicates with different simulation condition IDs?).

Reply:
This is probably the most likely scenario. However, this would also occur for example if different time points in the same line would have a different simulation condition id (not sure if this is a scenario that can occur)